indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionIntervalSpec.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.common.task;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.JodaUtils;
+import org.apache.druid.segment.SegmentUtils;
+import org.apache.druid.timeline.DataSegment;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Specifying an interval to compact. A hash of the segment IDs can be optionally provided for segment validation.
+ */
+public class CompactionIntervalSpec implements CompactionInputSpec
+{
+  public static final String TYPE = "interval";
+
+  private final Interval interval;
+  @Nullable
+  private final String sha256OfSortedSegmentIds;
+
+  @JsonCreator
+  public CompactionIntervalSpec(
+      @JsonProperty("interval") Interval interval,
+      @JsonProperty("sha256OfSortedSegmentIds") @Nullable String sha256OfSortedSegmentIds
+  )
+  {
+    if (interval != null && interval.toDurationMillis() == 0) {
+      throw new IAE("Interval[%s] is empty, must specify a nonempty interval", interval);
+    }
+    this.interval = interval;
+    this.sha256OfSortedSegmentIds = sha256OfSortedSegmentIds;
+  }
+
+  @JsonProperty
+  public Interval getInterval()
+  {
+    return interval;
+  }
+
+  @Nullable
+  @JsonProperty
+  public String getSha256OfSortedSegmentIds()
```

Comment:
nit: not personally super into this name, i feel like it makes me unnecessarily care about how the segment ids were hashed, but doesn't really matter i guess since this isn't so much directly used or directly constructed by users.